tools/cell_census_builder/tests/test_builder.py

```diff
@@ -1,26 +1,189 @@
+                "cell_type_ontology_term_id": "CL:0000192",
+                "assay_ontology_term_id": "EFO:0008720",
+                "disease_ontology_term_id": "PATO:0000461",
+                "organism_ontology_term_id": "NCBITaxon:9606",  # TODO: add one that fails the filter
+                "sex_ontology_term_id": "unknown",
+                "tissue_ontology_term_id": "CL:0000192",
+                "is_primary_data": False,
+                "self_reported_ethnicity_ontology_term_id": "na",
+                "development_stage_ontology_term_id": "MmusDv:0000003",
+                "donor_id": "donor_2",
+                "suspension_type": "na",
+                "assay": "test",
+                "cell_type": "test",
+                "development_stage": "test",
+                "disease": "test",
+                "self_reported_ethnicity": "test",
+                "sex": "test",
+                "tissue": "test",
+                "organism": "test",
+            }
         )
-        s_df.write(pa.Table.from_pandas(pd_df, preserve_index=False))
+        obs = obs_dataframe
+
+        # Create vars
+        feature_name = pd.Series(data=["a", "b", "c", "d"])
+        var_dataframe = pd.DataFrame(
+            data={
+                "feature_biotype": "gene",
+                "feature_is_filtered": False,
+                "feature_name": "ERCC-00002 (spike-in control)",
+                "feature_reference": "NCBITaxon:9606",
+            },
+            index=feature_name,
+        )
+        var = var_dataframe
+
+        # Create embeddings
+        random_embedding = np.random.rand(4, 4)
+        obsm = {"X_awesome_embedding": random_embedding}
+
+        # Create uns corpora metadata
+        uns = {}
+        uns["batch_condition"] = np.array(["a", "b"], dtype="object")
+
+        # Need to carefully set the corpora schema versions in order for tests to pass.
+        uns["schema_version"] = "3.0.0"  # type: ignore
+
+        return anndata.AnnData(X=X, obs=obs, var=var, obsm=obsm, uns=uns)
+
+    @pytest.fixture
+    def datasets(self, h5ad: anndata.AnnData) -> List[Dataset]:
+        first_h5ad = h5ad
+        second_h5ad = h5ad
+
+        first_h5ad_path = f"{self.assets_path}/first.h5ad"
+        second_h5ad_path = f"{self.assets_path}/second.h5ad"
+
+        first_h5ad.write_h5ad(first_h5ad_path)
+        second_h5ad.write_h5ad(second_h5ad_path)
+
+        return [
+            Dataset(dataset_id="first_id", corpora_asset_h5ad_uri="mock", dataset_h5ad_path=first_h5ad_path),
+            Dataset(dataset_id="second_id", corpora_asset_h5ad_uri="mock", dataset_h5ad_path=second_h5ad_path),
+        ]
+
+    def test_base_builder_creation(self, datasets: List[Dataset]) -> None:
+        """
+        Runs the builder, queries the census and performs a set of base assertions.
+        """
+
+        mock_prepare_file_system = patch("tools.cell_census_builder.__main__.prepare_file_system")
+        mock_prepare_file_system.start()
```

Comment:
Where are you stopping the patches?

Reply:
I suggest using a context manager or decorator to make sure the patch is cleaned up.

Reply:
you don't have to double nest. You can do this:
``` python
with patch("tools.cell_census_builder.__main__.prepare_file_system") as mock_prepare_file_system, patch(
            "tools.cell_census_builder.__main__.build_step1_get_source_assets", return_value=datasets
        ) as mock_get_assets:
   pass
```

Reply:
decorators also work.

Reply:
Ah nice, let me change to that. Unfortunately decorators conflict with pytest fixtures so we'll have to go with this.